worked_examples/geospatial/workshop_april_2024/Workshop_AprilMay2024_extra.ipynb

```diff
@@ -0,0 +1,446 @@
+   "execution_count": null,
+   "id": "a0d31f88",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import mapreader\n",
+    "assert mapreader.__version__ == '1.2.0'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f65f140c",
+   "metadata": {},
+   "source": [
+    "-------------"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "6151ccfd",
+   "metadata": {},
+   "source": [
+    "# Classify"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "0433e1a7",
+   "metadata": {},
+   "source": [
+    "We can also use our ``ClassifierContainer`` to infer labels on new datasets with a pre-trained/fine-tuned model.\n",
+    "\n",
+    "We are going to look at how to use our railspace model (https://huggingface.co/Livingwithmachines/mr_resnest101e_finetuned_OS_6inch_2nd_ed_railspace) to predict labels on patches.\n",
+    "To do this, we need to load the model from huggingface and then pass it in as the ``model`` argument in our ``ClassifierContainer``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e6aa7339",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    import timm\n",
+    "except ImportError: # timm isn't installed by default, so you might need to install it\n",
+    "    !pip install timm\n",
+    "    import timm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8586785c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_model = timm.create_model(\"hf_hub:Livingwithmachines/mr_resnest101e_finetuned_OS_6inch_2nd_ed_railspace\", pretrained=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b363ac68",
+   "metadata": {},
+   "source": [
+    "Now we set up the ``ClassifierContainer``"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f6276606",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mapreader import ClassifierContainer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5a0ca8ad",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_classifier = ClassifierContainer(\n",
+    "    model=my_model,\n",
+    "    labels_map={0:\"no\", 1:\"railspace\"}, # manually set this, you can find it in the model card - https://huggingface.co/Livingwithmachines/mr_resnest101e_finetuned_OS_6inch_2nd_ed_railspace\n",
+    "    dataloaders=None,\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "f95630bc",
+   "metadata": {},
+   "source": [
+    "The model can now be used to infer, or predict, the labels of \"unseen\" patches."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "edaabe9e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "patch_df = pd.read_excel(\"./patch_df.csv\", index_col=0)  # load our patch_df.csv file\n",
```

Comment:
Similar to the `read_excel` call in part2, I had to replace this with `read_csv`.